src/js/view/controls/tizen/tizen-controls.ts

```diff
@@ -98,40 +103,119 @@ class TizenControls extends Controls {
 
         this.div.appendChild(controlbar.element());
 
+        // Seekbar
+        const seekbar = this.seekbar = new TizenSeekbar(model, api, this.controlbar.elements.time);
+        this.div.appendChild(seekbar.element());
+
+        const enterSeekMode = () => {
+            if (this.seekbar) {
+                addClass(this.playerContainer, 'jw-flag-seek');
+                this.seekState = true;
+                this.seekbar.show();
+                api.pause();
+            }
+        };
+
+        const exitSeekMode = () => {
+            if (this.seekbar) {
+                removeClass(this.playerContainer, 'jw-flag-seek');
+                this.seekState = false;
+                this.seekbar.hide();
+            }
+        };
+
+        const updateSeek = (increment: number) => {
+            if (this.seekbar) {
+                this.seekbar.update(increment);
+            }
+        };
+
+        const seek = () => {
+            if (this.seekbar) {
+                this.seekbar.seek();
+                exitSeekMode();
+                api.play();
+                this.userInactive();
+            }
+        };
```

Comment:
What's the reason for creating these functions dynamically in `enable`?

Is it for `api` access?

It seems like it would be better to create private methods that access `this.api`, where `enable` and `disable` set and unset `this.api`, and we can use that as a proxy to tell if controls are enabled and exit early from those methods if not.

```
public get enabled (): boolean {
    return !!this.api;
}

private enterSeekMode (): void {
    if (!this.enabled || !this.seekbar) {
        return;
    }
    addClass(this.playerContainer, 'jw-flag-seek');
    this.seekState = true;
    this.seekbar.show();
    this.api.pause();
}

Reply:
I like that. Yeah, I had made them in enable to access `api`.